File: src/main/java/com/company/commitet_jm/service/file/FileService.java
package com.company.commitet_jm.service.file;

import com.company.commitet_jm.entity.FileCommit;
import com.company.commitet_jm.entity.Platform;
import com.company.commitet_jm.entity.TypesFiles;
import com.company.commitet_jm.component.ShellExecutor;

import java.io.File;
import java.util.List;

public interface FileService {
    void saveFileCommit(String baseDir, List<FileCommit> files, Platform platform);
    File correctPath(String baseDir, TypesFiles type);
    List<File> findBinaryFilesFromGitStatus(String repoDir, ShellExecutor executor);
    void unpackFiles(List<GitWorker.Pair<String, String>> files, Platform platform, ShellExecutor executor, String baseDir);
}

File: src/main/kotlin/com/company/commitet_jm/service/file/FileService.kt
package com.company.commitet_jm.service.file

import com.company.commitet_jm.entity.FileCommit
import com.company.commitet_jm.entity.Platform
import com.company.commitet_jm.entity.TypesFiles
import java.io.File

interface FileService {
    fun saveFileCommit(baseDir: String, files: MutableList<FileCommit>, platform: Platform)
    fun correctPath(baseDir: String, type: TypesFiles): File
    fun findBinaryFilesFromGitStatus(repoDir: String, executor: com.company.commitet_jm.component.ShellExecutor): List<File>
    fun unpackFiles(files: List<Pair<String, String>>, platform: Platform, executor: com.company.commitet_jm.component.ShellExecutor, baseDir: String)
}